ui-input-state/src/pointer_state.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright 2025 the UI Events Authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+#![allow(missing_docs)]
+
+use alloc::vec::Vec;
+use ui_events::pointer::{PointerButton, PointerButtons, PointerEvent};
+
+/// A stateful view of the primary pointer.
+#[derive(Clone, Debug, Default)]
+pub struct PointerState {
+    events: Vec<PointerEvent>,
+    pub down: PointerButtons,
+}
+
+impl PointerState {
+    /// Return `true` if the `button` was pressed within the last frame.
+    ///
+    /// This corresponds to having received a [`PointerEvent::Down`] event
+    /// for that button on the primary pointing device.
+    pub fn just_pressed(&self, button: PointerButton) -> bool {
+        self.events.iter().any(|e| {
+            matches!(e, &PointerEvent::Down { button: b, pointer: pi, .. } if b == Some(button) && pi.is_primary_pointer())
+        })
+    }
+
+    pub fn just_released(&self, button: PointerButton) -> bool {
+        self.events
+            .iter()
+            .any(|e| matches!(e, &PointerEvent::Up { button: b, .. } if b == Some(button)))
```

Comment:
I agree that this should also be guarded to be on the primary pointer, possibly by moving the check up one level...